middleware/common/include/common/message/coordinate.h

```diff
@@ -93,24 +93,23 @@ namespace casual
                   return done();
                }
 
-               template< typename I>
-               auto failed( I&& id) -> std::vector< strong::correlation::id>
+               template< typename I, typename Predicate>
+               auto failed( I&& id, Predicate predicate) -> std::vector< strong::correlation::id>
                {
-                  auto has_id_and_pending = [ &id]( auto& pending){ return pending.id == id && pending.state == Pending::State::pending;};
-                  auto update_failed_and_transform_correlation =  []( auto& pending)
-                  {
-                     pending.state = Pending::State::failed; 
-                     return pending.correlation;
-                  };
-
-                  return algorithm::transform_if( m_pending, update_failed_and_transform_correlation, has_id_and_pending);
+                  return algorithm::transform_if( m_pending, []( auto& pending)
+                     {
+                        pending.state = Pending::State::failed;
+                        return pending.correlation;
+                     },
+                     predicate
+                  );
                }
```

Comment:
`id` is not used. Either remove it from the arguments, or invoke the predicate with it. If you let the predicate take the `id` as a parameter you don't need to capture it in the lambda (creation of the predicate) further down. I prefer the latter, but the former is good too.